apache/superset
# fix(date_parser): enhance date parsing for time range filters

### SUMMARY
This PR addresses the date parsing issues related to time range filters as reported in [Issue #30592](https://github.com/apache/superset/issues/30592). The changes made to the `date_parser.py` file enhance the handling of various date formats, including relative dates and specific time components. The `parse_human_datetime` function has been updated to correctly parse inputs such as "today, HH:MM:SS" and "today, HH:MM".

fixes: https://github.com/apache/superset/issues/30592

### TESTING INSTRUCTIONS
1. Run the unit tests located in `tests/unit_tests/utils/date_parser_tests.py` to verify that all tests pass.

<img width="737" alt="Screenshot 2024-12-10 at 3 41 55 PM" src="https://github.com/user-attachments/assets/fb6f8bc6-5d34-4a15-81c1-9925efa98641">


2. Manually test the following date parsing scenarios:
   - "today"
   - "today, 11:30"
   - "today, 11:30:30"
   - "1 hour ago"
   - "this year"
   - "year"
   - "monday", "tuesday", etc.


### ADDITIONAL INFORMATION
- [x] Has associated issue: [Issue #30592](https://github.com/apache/superset/issues/30592)
